src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -397,21 +397,25 @@ protected final void verify(Checker checker,
      * @param testConfig hardcoded test config.
      * @param parsedConfig parsed config from input file.
      */
-    private static void verifyConfig(Configuration testConfig, Configuration parsedConfig) {
+    private static void verifyConfig(Configuration testConfig, Configuration parsedConfig,
+                                     InputConfiguration inputConfiguration)
+            throws CheckstyleException {
         assertWithMessage("Check name differs from expected.")
                 .that(testConfig.getName())
                 .isEqualTo(parsedConfig.getName());
-        assertWithMessage("Property keys differ from expected.")
-                .that(parsedConfig.getAttributeNames())
-                .isEqualTo(testConfig.getAttributeNames());
         for (String attribute : testConfig.getAttributeNames()) {
-            try {
+            if(inputConfiguration.isDefaultAttribute(attribute)) {
+                assertWithMessage("Property value for key %s differs from expected.", attribute)
+                        .that(inputConfiguration.getDefaultAttributeValue(attribute))
+                        .isEqualTo(testConfig.getAttribute(attribute));
+            }
+            else if(inputConfiguration.isNonDefaultAttribute(attribute)) {
                 assertWithMessage("Property value for key %s differs from expected.", attribute)
                         .that(parsedConfig.getAttribute(attribute))
                         .isEqualTo(testConfig.getAttribute(attribute));
             }
```

Comment:
we miss case when `parsedConfig` contains some redundant property which does not exist in `testConfig `

Reply:
can we do in the following way?

1. check that all properties from `parsedConfig` exist in `testConfig` and have same value
2. get properties that only exist in `testConfig` but not in `parsedConfig` and compare them with `inputConfiguration.defaultProperties`

Reply:
done

Reply:
hmm, can we avoid using `inputConfiguration.isNonDefaultAttribute` and `inputConfiguration.isDefaultAttribute` methods since we can distinguish default and non default attributes without them

smth like:
```
        for (String attribute : parsedConfig.getAttributeNames()) {
            assertWithMessage("Property value for key %s differs from expected.", attribute)
                    .that(testConfig.getAttribute(attribute))
                    .isEqualTo(parsedConfig.getAttribute(attribute));
        }

        for (String attribute : onlyInTestConfigProperties) {
            assertWithMessage("Property value for key %s differs from expected.", attribute)
                    .that(inputConfiguration.getDefaultAttributeValue(attribute))
                    .isEqualTo(testConfig.getAttribute(attribute));
        }
```

Reply:
done